flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/FsStateChangelogWriter.java

```diff
@@ -167,6 +167,20 @@ class FsStateChangelogWriter implements StateChangelogWriter<ChangelogStateHandl
         this.localChangelogRegistry = localChangelogRegistry;
     }
 
+    @Override
+    public void append(byte[] value) throws IOException {
+        LOG.trace("append metadata to {}: {} bytes", logId, value.length);
+        checkState(!closed, "%s is closed", logId);
+        activeChangeSet.add(new StateChange(value));
+        activeChangeSetSize += value.length;
+        if (activeChangeSetSize >= preEmptivePersistThresholdInBytes) {
+            LOG.debug(
+                    "pre-emptively flush {}MB of appended changes to the common store",
+                    activeChangeSetSize / 1024 / 1024);
+            persistInternal(notUploaded.isEmpty() ? activeSequenceNumber : notUploaded.firstKey());
+        }
+    }
```

Comment:
There are a lot of duplicate code between the two "append" methods.

Please extract the two common part and rename this to "appendMeta" for example.